src/collections/news/news.json

```diff
@@ -1,4 +1,4 @@
 {
     "layout": "layouts/post",
-    "permalink": "/news/{{ date | slugify }}-{{ slug | default(title) | slugify }}/"
+    "permalink": "/news/{{ date | dateString | slugify }}-{{ slug | default(title) | slugify }}/"
```

Comment:
I don't think slugify needs to be applied to the date after your filter, it should always be YYYY-MM-DD.